python/dlr/dlr_model.py

```diff
@@ -87,16 +70,47 @@ def _lazy_init_output_shape(self):
 
     def _parse_backend(self):
         backend = c_char_p()
-        _check_call(_LIB.GetDLRBackend(byref(self.handle),
+        self._check_call(self._lib.GetDLRBackend(byref(self.handle),
                                        byref(backend)))
         return backend.value.decode('ascii')
 
     def _get_version(self):
         version = c_char_p()
-        _check_call(_LIB.GetDLRVersion(byref(version)))
+        self._check_call(self._lib.GetDLRVersion(byref(version)))
         return version.value.decode('ascii')
 
-    def __init__(self, model_path, dev_type='cpu', dev_id=0):
+    def _check_call(self, ret):
+        """
+        Check the return value of C API call
+        This function will raise exception when error occurs.
+        Wrap every API call with this function
+
+        Parameters
+        ----------
+        ret : int
+            return value from API calls
+        """
+        if ret != 0:
+            raise DLRError(self._lib.DLRGetLastError().decode('ascii'))
+
+    def _init_libdlr(self):
+        paths = find_lib_path()
+        
+        local_libdlr_so_path = os.path.join(self.model_path, "libdlr.so")
+        if sys.platform == 'win32':
+            local_libdlr_so_path = os.path.join(self.model_path, "dlr.dll")
+        elif sys.platform == 'darwin':
+            local_libdlr_so_path = os.path.join(self.model_path, "dlr.dylib")
+        
+        if not self.use_default_dlr and os.path.exists(local_libdlr_so_path):
+            self.logger.info(f"Found libdlr.so in model artifact. Using {local_libdlr_so_path}")
```

Comment:
Lets avoid f-string formatting because we have to support python 3.5 for some devices.

Reply:
Good catch 👍